File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/model/inheritance/AdministratorMetadataBase.java
package org.babyfish.jimmer.sql.model.inheritance;

import org.babyfish.jimmer.pojo.Static;
import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

@MappedSuperclass
public interface AdministratorMetadataBase extends NamedEntity {

    String getEmail();

    String getWebsite();

    @OneToOne(inputNotNull = true)
    @Nullable
    @OnDissociate(DissociateAction.DELETE)
    @Static(alias = "default", name="administratorId", idOnly = true)
    @Static(alias = "composite")
    Administrator getAdministrator();
}

File: project/jimmer-sql-kotlin-test-model/src/main/kotlin/org/babyfish/jimmer/sql/kt/model/inheritance/AdministratorMetadataBase.kt
package org.babyfish.jimmer.sql.kt.model.inheritance

import org.babyfish.jimmer.sql.*

@MappedSuperclass
interface AdministratorMetadataBase : NamedEntity {

    val email: String

    val website: String

    @Key
    @OneToOne(inputNotNull = true)
    @OnDissociate(DissociateAction.DELETE)
    val administrator: Administrator?
}